### ISSUE TITLE
Lesson-1 cabal install

### ISSUE THREAD

**Issue by Morkrom** [ryantrinkle/memoise#2]
I get this output on cabal install after cloning lesson 1. Is it because I have Snap installed?

cabal install
Resolving dependencies...
cabal: Could not resolve dependencies:
trying: memoise-0.1 (user goal)
trying: base-4.7.0.1/installed-df2... (dependency of memoise-0.1)
next goal: snap (dependency of memoise-0.1)
rejecting: snap-0.14.0/installed-6a2..., 0.13.3.2/installed-c1f..., 0.14.0,
0.13.3.2, 0.13.3.1, 0.13.3, 0.13.2.9, 0.13.2.8, 0.13.2.7, 0.13.2.6, 0.13.2.5,
0.13.2.4, 0.13.2.3, 0.13.2.2, 0.13.2.1, 0.13.2.0, 0.13.1.2, 0.13.1.1, 0.13.1,
0.13.0.4, 0.13.0.3, 0.13.0.2, 0.13.0.1, 0.13.0 (conflict: memoise =>
snap>=0.12 && <0.13)
trying: snap-0.12.1
next goal: template-haskell (dependency of snap-0.12.1)
rejecting: template-haskell-2.9.0.0/installed-1b1..., 2.9.0.0 (conflict: snap
=> template-haskell>=2.2 && <2.9)
rejecting: template-haskell-2.8.0.0 (conflict: base==4.7.0.1/installed-df2...,
template-haskell => base==4.6._)
rejecting: template-haskell-2.7.0.0 (conflict: base==4.7.0.1/installed-df2...,
template-haskell => base==4.5._)
rejecting: template-haskell-2.6.0.0 (conflict: base==4.7.0.1/installed-df2...,
template-haskell => base==4.4._)
rejecting: template-haskell-2.5.0.0 (conflict: base==4.7.0.1/installed-df2...,
template-haskell => base==4.3._)
rejecting: template-haskell-2.4.0.1 (conflict: base==4.7.0.1/installed-df2...,
template-haskell => base==4.2.*)
rejecting: template-haskell-2.4.0.0 (conflict: base==4.7.0.1/installed-df2...,
template-haskell => base>=3 && <4.3)
rejecting: template-haskell-2.3.0.1, 2.3.0.0, 2.2.0.0 (conflict:
base==4.7.0.1/installed-df2..., template-haskell => base<4.3)
Dependency tree exhaustively searched.

**Comment by mightybyte** [comment]
It's probably because you're using GHC 7.8, which wasn't released yet when memoi.se was written.  It still works fine for me with GHC 7.6.

**Comment by ryantrinkle** [comment]
I agree; this is probably a 7.8 issue.  @Morkrom, if you try loosening or removing the upper version bounds in the cabal file, that may help.  If that works for you, please let me know (or send a PR) so I can update the repo.

**Comment by Morkrom** [comment]
That did it, @ryantrinkle. Thanks for input, @mightybyte.
